lib/dlss/docker/capistrano/tasks/docker_compose.rake

```diff
@@ -0,0 +1,198 @@
+# frozen_string_literal: true
+
+# Capistrano plugin hook to set default values
+namespace :load do
+  task :defaults do
+    set :docker_compose_file, fetch(:docker_compose_file, 'docker-compose.yml')
+
+    set :docker_compose_migrate_use_hooks, fetch(:docker_compose_migrate_use_hooks, true)
+    set :migration_role, fetch(:migration_role, :db)
+    set :migration_servers, -> { primary(fetch(:migration_role)) }
+    set :migration_command, fetch(:migration_command, 'db:migrate')
+
+    set :docker_compose_seed_use_hooks, fetch(:docker_compose_seed_use_hooks, false)
+    set :seed_command, fetch(:migration_command, 'db:seed')
+
+    set :docker_compose_build_use_hooks, fetch(:docker_compose_build_use_hooks, true)
+    set :build_roles, fetch(:build_roles, [:app])
+    set :dereference_files, fetch(:dereference_files)
+    set :dereference_dirs, fetch(:dereference_dirs, [])
+
+    set :docker_compose_rabbitmq_use_hooks, fetch(:docker_compose_rabbit_use_hooks, false)
+    set :rabbitmq_role, fetch(:rabbitmq_role, :app)
+    set :rabbitmq_servers, -> { primary(fetch(:rabbitmq_role)) }
+    set :rabbitmq_command, fetch(:rabbitmq_command, 'rabbitmq:setup')
+
+    set :docker_compose_restart_use_hooks, fetch(:docker_compose_restart_use_hooks, true)
+
+    set :copy_assets_role, fetch(:copy_assets_role, :app)
+    set :assets_path, fetch(:assets_path, 'public/assets')
+    set :docker_compose_copy_assets_use_hooks, fetch(:docker_compose_copy_assets_use_hooks, true)
+  end
+end
+
+# Integrate hooks into Capistrano
+namespace :deploy do
+  before :starting, :add_docker_hooks do
+    invoke 'docker_compose:add_migrate_hooks' if fetch(:docker_compose_migrate_use_hooks)
+    invoke 'docker_compose:add_seed_hooks' if fetch(:docker_compose_seed_use_hooks)
+    invoke 'docker_compose:add_build_hooks' if fetch(:docker_compose_build_use_hooks)
+    invoke 'docker_compose:add_rabbitmq_hooks' if fetch(:docker_compose_rabbitmq_use_hooks)
+    invoke 'docker_compose:add_restart_hooks' if fetch(:docker_compose_restart_use_hooks)
+    invoke 'docker_compose:add_copy_assets_hooks' if fetch(:docker_compose_copy_assets_use_hooks)
+  end
+end
+
+namespace :docker_compose do
+  task :add_migrate_hooks do
+    after 'deploy:publishing', 'docker_compose:migrate'
+  end
+
+  task :add_seed_hooks do
+    after 'deploy:publishing', 'docker_compose:seed'
+  end
+
+  task :add_build_hooks do
+    after 'deploy:updating', 'docker_compose:build'
```

Comment:
It's probably fine, but the `deploy:updating` hook is only used for deploys not rollbacks. I'm not sure how far down the rollback road you've gone, so this may be a thing to file away for another day. Assuming we'd want to re-build in a rollback scenario, of course. If we do, we could add `after 'deploy:reverting', 'docker_compose:build'`. (The same applies to the shared_configs hook below, but since we presumably haven't been bitten by this before, you may be able to ignore what I'm typing here entirely.)